### PR TITLE
mining resource (mintable & burnable) currency system

### PR DESCRIPTION
**Pull request by justinphamnz**
- [x] Allow council account sets to mint/burn mining resource
- [x]  Deposit and withdraw system to transfer mining resources from hot wallet to chain storage.
- [x] Allow mining resource liquidity on dex pool

### CODE REVIEW

File: pallets/mining/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+    fn do_deposit(
+        who: T::AccountId,
+        from: T::AccountId,
+        amount: Balance,
+    ) -> DispatchResult {
+        if amount.is_zero() {
+            return Ok(());
+        }
+
+        ensure!(
```
Comment (jerryjren) [inline pallets/mining/src/lib.rs:305]: Hi Justin, I'm not sure about the is_mining_origin part. Does this mean only allow mining_origin could deposit and withdraw?

File: pallets/mining/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+        who: T::AccountId,
+        dest: T::AccountId,
+        amount: Balance,
+    ) -> DispatchResult {
+        if amount.is_zero() || who == dest {
+            return Ok(());
+        }
+
+        ensure!(
+            Self::is_mining_origin(&who),
```
Comment (jerryjren) [inline pallets/mining/src/lib.rs:333]: Same comment as do_deposit

File: pallets/mining/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+            Error::<T>::NoPermission);
+
+        let mining_treasury = Self::bit_mining_resource_account_id();
+        ensure!(
```
Comment (jerryjren) [inline pallets/mining/src/lib.rs:337]: When withdraw bit to blockchain, I think we should check the balance in the centralized place. i.e. bit country database

**Review by jerryjren — COMMENTED** [review]
Thanks Justin, all looks good. Just a few minor things. Feel free to correct me, I probably wrong about the withdraw/deposit concept.

File: pallets/mining/src/lib.rs (hunk 1 above)
Comment (justinphamnz) [reply pallets/mining/src/lib.rs:305]: That's right. Only certain origin can trigger deposit and withdraw.

File: pallets/mining/src/lib.rs (hunk 3 above)
Comment (justinphamnz) [reply pallets/mining/src/lib.rs:337]: All extrinsic calls from these method should be validated on offchain process before triggering these calls.

**Review by jerryjren — APPROVED** [review]

**Pull request merged**